extractor/src/main/java/org/schabi/newpipe/extractor/services/youtube/YoutubeDashManifestCreator.java

```diff
@@ -0,0 +1,1605 @@
+            segmentCount = responseHeaders.get("X-Head-Seqnum").get(0);
+        } catch (final IndexOutOfBoundsException e) {
+            throw new YoutubeDashManifestCreationException(
+                    "Could not generate the DASH manifest: could not get the value of the X-Head-Time-Millis or the X-Head-Seqnum header of the post live DVR streaming URL", e);
+        }
+
+        if (isNullOrEmpty(segmentCount)) {
+            throw new YoutubeDashManifestCreationException(
+                    "Could not generate the DASH manifest: could not get the number of segments of the post live DVR stream");
+        }
+
+        final Document document = generateDocumentAndMpdElement(new String[]{streamDuration},
+                DeliveryType.LIVE, itagItem, durationSecondsFallback);
+        generatePeriodElement(document);
+        generateAdaptationSetElement(document, itagItem);
+        generateRoleElement(document);
+        generateRepresentationElement(document, itagItem);
+        if (itagItem.itagType == ItagItem.ItagType.AUDIO) {
+            generateAudioChannelConfigurationElement(document, itagItem);
+        }
+        generateSegmentTemplateElement(document, postLiveStreamDvrStreamingUrl, DeliveryType.LIVE);
+        generateSegmentTimelineElement(document);
+        generateSegmentElementForPostLiveDvrStreams(document, targetDurationSec, segmentCount);
+
+        return buildResult(originalPostLiveStreamDvrStreamingUrl, document,
+                POST_LIVE_STREAMS_MANIFESTS_GENERATED);
+    }
+
+    /**
+     * Create DASH manifests from a YouTube progressive stream.
+     *
+     * <p>
+     * Progressive is one of the YouTube DASH streams which works with range requests and without
+     * the need to get a manifest.
+     * </p>
+     *
+     * <p>
+     * They can be found on all videos, and for all streams on videos which comes from a YouTube
+     * partner or for videos with a big amount of views.
+     * </p>
+     *
+     * <p>This method needs:
+     *     <ul>
+     *         <li>the base URL of the stream (which returns the whole stream, after redirects and
+     *         if the URL is valid, if you try to access to it);</li>
+     *         <li>an {@link ItagItem} which needs to contain the following information:
+     *              <ul>
+     *                  <li>its type (see {@link ItagItem.ItagType}), to identify if the content is
+     *                  an audio or a video stream</li>
+     *                  <li>its bitrate;</li>
+     *                  <li>its mime type;</li>
+     *                  <li>its codec(s);</li>
+     *                  <li>for audio streams: its audio channels;</li>
+     *                  <li>for video streams: its width and height.</li>
+     *              </ul>
+     *         </li>
+     *         <li>the duration of the video, which will be used if the duration could not be
+     *         get from the {@link ItagItem}.</li>
```

Comment:
You used 'parsed' everywhere else.

```suggestion
     *         parsed from the {@link ItagItem}.</li>
```